src/linear_algebra/general_linear_group.lean

```diff
@@ -191,6 +193,22 @@ lemma to_GL_pos_injective :
 (show function.injective ((coe : GL_pos n R → matrix n n R) ∘ to_GL_pos),
  from subtype.coe_injective).of_comp
 
+lemma coe_to_GL_pos_ext (g : special_linear_group n R) (i j : n) : g i j = (g : (GL_pos n R)) i j :=
+rfl
+
+@[simp]
+lemma coe_to_GL_pos_det (g : special_linear_group n R) : det (g : GL_pos n R) = 1 :=
+g.prop
+
+@[simp]
+lemma coe_coe_matrix (g : special_linear_group n ℤ) (i j : n) :
+  ((g : special_linear_group n R) : GL_pos n R) i j = (g : matrix n n ℤ) i j := rfl
```

Comment:
I think what you actually want here is:
```lean
/-- Coercing a `special_linear_group` via `GL_pos` and `GL` is the same as coercing striaght to a matrix -/
@[simp]
lemma coe_GL_pos_coe_GL_coe_matrix (g : special_linear_group n R) :
  (↑(↑(↑(g : special_linear_group n R) : GL_pos n R) : GL n R) : matrix n n R) = ↑g := rfl
```
This lemma makes your lemma follow trivially

Reply:
Aha yes! I see what you mean now. Ok I've added this and removed the other lemma. Should I merge #12397 into this PR and then wait until that is added?

Reply:
So I merged #12397 into this but it not happy with `coe_to_GL_pos_ext`  and `coe_to_GL_pos_det`.  it keeps saying "function expected at
  g
term has type
  {A // A.det = 1}"

Any idea how to fix this?

Reply:
ok I made an ugly fix to this, but there is probably not the right thing to do.